sky/spot/spot_state.py

```diff
@@ -18,9 +19,25 @@
 _DB_PATH = _DB_PATH.expanduser().absolute()
 _DB_PATH.parents[0].mkdir(parents=True, exist_ok=True)
 
+# Module-level connection/cursor; thread-safe as the module is only imported
+# once.
 _CONN = sqlite3.connect(str(_DB_PATH))
 _CURSOR = _CONN.cursor()
 
+
+@contextlib.contextmanager
+def _safe_cursor():
```

Comment:
It seems we have this in the `sky.skylet.configs` as well. Do we want to refactor it as a common utility function in `sky.utils.db_utils`?

Also, we have a `db_utils.SQLiteConn` which creates a thread-level connection for the database and serves for the same purpose in `job_lib` and `global_user_state`. Can we have choose one of them for all the modules using sqlite?

Reply:
Good point - moved it into `db_utils.safe_cursor()`. `SQLiteConn` doesn't come with auto-commit as a context manager so I didn't use it.